pkg/newlog/cmd/newlogd.go

```diff
@@ -568,24 +568,30 @@ func getMemlogMsg(logChan chan inputEntry, panicFileChan chan []byte) {
 			continue
 		}
 		sourceName, msgTime, origMsg := getSourceFromMsg(string(bytes))
-		logInfo, ok := agentlog.ParseLoginfo(origMsg)
 
 		var pidStr string
 		var isApp bool
-
-		if strings.Contains(string(bytes), "guest_vm") {
-			logmetrics.AppMetrics.NumInputEvent++
+		var jsonOK bool
+		var logInfo agentlog.Loginfo
+		if strings.Contains(sourceName, "guest_vm") {
```

Comment:
Would not it be better to check the `sourceName` against the `contains('guest_vm')` and not the whole log string? Because that what we are interested in after all: to filter out the 'guest_vm' source, which is exactly the the meaning of the parsed field 'sourceName'. Because if there is a log message which contains the 'guest_vm' string (regardless of the real source of the log message), it will be accounted as a vm source.

Reply:
but this is exactly what I'm doing, and that is exactly what it's explained in the commit message :smile:

Reply:
same as above. sorry for the noise.